### PR TITLE
Refactor code and add getData method

### PR DESCRIPTION
**Pull request by j10a1n15**
This pull request includes the following changes:

- Fix error handling in mods.js

- Add general getData() method in mods.js

- Renamed sendRequest to sendCommitRequest

- Refactor code to use SystemUtils for fetching data

- Refactor main_menu.js to use SystemUtils for fetching data

- Add empty line in main_menu.js

### CODE REVIEW

**Comment by Su386yt** [comment]
Okay I should have wrote this on the others ones but,, stop using AI generated prs

**Comment by j10a1n15** [comment]
> Okay I should have wrote this on the others ones but,, stop using AI generated prs

why D:

**Comment by Su386yt** [comment]
I gotta read them and half of them just don't make sense

**Comment by j10a1n15** [comment]
okayy

File: src/commands/mods.js [hunk 1]
```diff
@@ -362,7 +362,8 @@ async function handleSearchCommand(client, interaction) {
     .setTitle("Mod - " + mod.name)
     .setDescription(`Mod ID: ${Object.keys(mods)[0]}\n\nDownload: ${mod.download}`)
     .addFields({ name: "Versions", value: versionsField })
-    .addFields({ name: "Beta Versions", value: betaVersionsField });
+    if (betaVersionsField)
```
Comment (j10a1n15) [inline src/commands/mods.js:365]: <@283738275552821248> @Su386yt here is it fixed lol

**Pull request merged**